gtsam/sfm/TranslationRecovery.cpp

```diff
@@ -71,36 +74,63 @@ TranslationRecovery::TranslationRecovery(
 NonlinearFactorGraph TranslationRecovery::buildGraph() const {
   NonlinearFactorGraph graph;
 
-  // Add all relative translation edges
+  // Add translation factors for input translation directions.
   for (auto edge : relativeTranslations_) {
     graph.emplace_shared<TranslationFactor>(edge.key1(), edge.key2(),
                                             edge.measured(), edge.noiseModel());
   }
 
+  // Add between factors for optional relative translations.
+  for (auto edge : params_.getBetweenTranslations()) {
+    Key k1 = getUniqueKey(edge.key1()), k2 = getUniqueKey(edge.key2());
+    if (k1 != k2) {
+      graph.emplace_shared<BetweenFactor<Point3>>(k1, k2, edge.measured(),
+                                                  edge.noiseModel());
+    }
+  }
+
   return graph;
 }
 
 void TranslationRecovery::addPrior(
-    const double scale, NonlinearFactorGraph *graph,
+    const double scale, const boost::shared_ptr<NonlinearFactorGraph> graph,
```

Comment:
I think `*` is right. In `.i` file use `@` to indicated an output parameter.

Reply:
changed all shared_ptr back to *, this change was to wrap these methods, but later I decided not to wrap them.

Reply:
This works in wrapper... Use @